crates/ruma-common/src/identifiers/client_secret.rs

```diff
@@ -24,7 +24,13 @@ impl ClientSecret {
     }
 }
 
-opaque_identifier_validated!(ClientSecret, ruma_identifiers_validation::client_secret::validate);
+owned_identifier!(OwnedClientSecret, ClientSecret);
+
+opaque_identifier_validated!(
+    ClientSecret,
+    OwnedClientSecret,
+    ruma_identifiers_validation::client_secret::validate
+);
```

Comment:
If we need to name the owned type here, we might as well call `owned_identifier!` from `opaque_identifier!` / `opaque_identifier_validated!` and have the owned macro be an implementation detail of the macros, basically.

Reply:
That'd hide the identifier from the code, and I've exposed it explicitly as needing to input the identifier in the macro to make it more clear that that identifier exists.

I'm not entirely partial to making the macro generate `Owned$id`, I'd rather have that part at least be explicit, if possible.